network-config-analyzer/CanonicalHyperCubeSet.py

```diff
@@ -402,11 +402,11 @@ def _prepare_common_active_dimensions(self, other):
         :return: result for 'other' (which may be copied and changed or not)
         """
         if self.active_dimensions == other.active_dimensions:
-            return other
+            return other.copy()
         required_active_dimensions = set(self.active_dimensions + other.active_dimensions)
         self._set_active_dimensions(required_active_dimensions)
         if set(other.active_dimensions) == required_active_dimensions:
-            return other
+            return other.copy()
```

Comment:
Changed the copy. It's not easy to create such a unit test, because after the 'or' operator the other object is still equal, it might change later, through changing the self.